File: VintedProject/src/main/java/it/unical/ea/VintedProject/dto/ServiceError.java
package it.unical.ea.VintedProject.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;


@Data
@NoArgsConstructor
@AllArgsConstructor
public class ServiceError {
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd-MM-yyyy hh:mm:ss")
    private Date timestamp;
    private String url;
    private String message;
}

File: VintedAndroid/app/src/main/java/com/example/vintedandroid/model/dto/ServiceError.kt
package com.example.vintedandroid.model.dto

import com.fasterxml.jackson.annotation.JsonFormat
import java.time.LocalDateTime
import java.util.Date

class ServiceError (
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd-MM-yyyy hh:mm:ss")
    val timestamp : LocalDateTime, // controllare se è meglio questo o Date
    val url : String,
    val message : String
        ){

}